manifests/init.pp

```diff
@@ -254,6 +254,7 @@
 
   if $enable_docker {
     include pulpcore::plugin::container
+    include foreman_proxy_content::container
```

Comment:
We know the exact pulp URL so it would be better to include the foreman_proxy::plugin::
```puppet
    class { 'foreman_proxy::plugin::container_gateway':
      pulp_url => "https://${servername}",
    }
```

This means that if Pulp runs on a URL that's different from the system hostname, it'll still function correctly. It makes it more predictable: there's an answer (IIRC set via `--certs-hostname` and stored in answers). If you change the hostname (like with `hostnamectl`) and rerun the installer, it'll still continue to serve with the old hostname. Only until you update the certificates will you change everything end to end.

If we follow this path, we can slowly make `katello-change-hostname` less needed.

Reply:
After adding in the class like you mentioned above and enabling the plugin, I'm getting a duplicate declaration error. Does this class declaration somehow conflict with what's in the answers file for foreman-proxy-content?

Reply:
Yes. I would suggest you don't include it in the answers file. We've done the same thing with the pulp plugin itself. This module is responsible for connecting them.

More background information: in Puppet there's a [roles and profiles method](https://puppet.com/docs/pe/2019.8/osp/the_roles_and_profiles_method.html). While it doesn't 100% map to our installer, it helps you understand a lot.

First of all, there are the component modules. They provide a base layer of functionality. Think apache, postgresql, redis. The end user of the installer doesn't directly see them but they're certainly there.

A layer up there are profile modules. They combine component modules in higher level functionality. This content module is that: they take a bunch of component modules and provide complete functionality. Another example is puppet-katello. We can also argue that puppet-foreman_proxy has some of those elements since it can manage a DHCP and DNS server as well.

In our installer we don't really have role modules. You could argue that an installer scenario is a role because it determines what can be enabled.

With that background in mind, let's apply it to the current case. We can expose all component modules to the user, but I'm arguing that instead we should bundle up the entire functionality in this module and only expose this module. We can still keep `foreman_proxy::plugin::container_gateway` and include it here, but the end user doesn't have to know about it. We can connect answers that we know that are related (`$pulp_url`) which also means they can't go out of sync.

Another way to look at it is whether it makes sense for a user to enable one part, but not the other. The container gateway plugin doesn't make a whole lot of sense without the other parts so exposing it via the answers file is probably not the right choice.

Reply:
> You could argue that an installer scenario is a role because it determines what can be enabled.

After reading about the differences between roles and profiles I think I agree with this too...

> Another way to look at it is whether it makes sense for a user to enable one part, but not the other. The container gateway plugin doesn't make a whole lot of sense without the other parts so exposing it via the answers file is probably not the right choice.

Yeah I see what you mean about keeping the container gateway more closely connected to the parameters that are already configured.  It seemed a little off to me to reconfigure what we already know. I'll update it.